src/trio/_util.py

```diff
@@ -327,6 +327,21 @@ def __call__(cls, *args: object, **kwargs: object) -> None:
     def _create(cls: type[T], *args: object, **kwargs: object) -> T:
         return super().__call__(*args, **kwargs)  # type: ignore
 
+    def __new__(cls, name: str, bases: tuple[type, ...], namespace: dict[str, Any], **kwargs: object) -> type:  # type: ignore[explicit-any]
+        old_reduce = namespace.get("__reduce__")
+
+        def __reduce__(self: object) -> str | tuple[Any, ...]:  # type: ignore[explicit-any]
+            if old_reduce is not None:
+                result = old_reduce(self)
+                assert not isinstance(result, str)
+            else:
+                result = super(type(self), self).__reduce__()  # type: ignore[misc]
+
+            return (type(self)._create, *result[1:])  # type: ignore[attr-defined]
+
+        namespace["__reduce__"] = __reduce__
+        return super().__new__(cls, name, bases, namespace, **kwargs)
```

Comment:
I'm a bit unsure if just tossing out the first object in the tuple is the right thing to do. But I'm not sure what else to do, so maybe just add a comment; or something like `assert result[0] is type(self)` (in the `if old_reduce is not None` branch) to check nobody's doing anything weird.

Reply:
Actually I think it's better to give up if there's already `__reduce__`.

Reply:
any reason to keep the new `__new__` at all?

Reply:
Just me forgetting that I added it in this PR. I'll remove it!